orchagent/fgnhgorch.cpp

```diff
@@ -0,0 +1,1500 @@
+#include <assert.h>
+#include <inttypes.h>
+#include "fgnhgorch.h"
+#include "routeorch.h"
+#include "logger.h"
+#include "swssnet.h"
+#include "crmorch.h"
+#include <array>
+
+extern sai_object_id_t gVirtualRouterId;
+extern sai_object_id_t gSwitchId;
+
+extern sai_next_hop_group_api_t*    sai_next_hop_group_api;
+extern sai_route_api_t*             sai_route_api;
+
+extern RouteOrch *gRouteOrch;
+extern CrmOrch *gCrmOrch;
+
+FgNhgOrch::FgNhgOrch(DBConnector *db, DBConnector *appDb, DBConnector *stateDb, vector<string> &tableNames, NeighOrch *neighOrch, IntfsOrch *intfsOrch, VRFOrch *vrfOrch) :
+        Orch(db, tableNames),
+        m_neighOrch(neighOrch),
+        m_intfsOrch(intfsOrch),
+        m_vrfOrch(vrfOrch),
+        m_stateWarmRestartRouteTable(stateDb, STATE_FG_ROUTE_TABLE_NAME),
+        m_routeTable(appDb, APP_ROUTE_TABLE_NAME)
+{
+     SWSS_LOG_ENTER();
+}
+
+
+/* calculate_bank_hash_bucket_start_indices: generates the hash_bucket_indices for all banks
+ * and stores it in fgNhgEntry for the group. 
+ * The function will identify the # of next-hops assigned to each bank and 
+ * assign the total number of hash buckets for a bank, based on the proportional
+ * number of next-hops in the bank. 
+ * eg: Bank0: 6 nh, Bank1: 3 nh, total buckets: 30 => 
+ *      calculate_bank_hash_bucket_start_indices: Bank0: Bucket# 0-19, Bank1: Bucket# 20-29
+ */
+void calculate_bank_hash_bucket_start_indices(FgNhgEntry *fgNhgEntry)
```

Comment:
>calculate_bank_hash_bucket_start_indices [](start = 5, length = 40)

CamelCases ?

Reply:
Will you also change this to CamelCases?

---
In reply to: [487340495](https://github.com/Azure/sonic-swss/pull/1315#discussion_r487340495) [](ancestors = 487340495)

Reply:
Oh, missed converting this one to camelCase, will do so